From 8a734652d8b03f31ed7b4f597ab2377eb66faa46 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Fri, 13 Jan 2023 07:31:43 +0200
Subject: [PATCH] [SW-115022] hl225b: rma mode: dont enable debug

On HL225B cards, no debug features are allowed.
Enabling RMA on HL225B card, should not disable security

BIT(25) of efuse indicates if the card is BIS complaint or not
If this bit is set, entering RMA debug mode is not allowed

Change-Id: Ieec6f5155b2bcbd6d809e3b2a47c66ea6385a7a2
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/app/ext_rom/Makefile b/app/ext_rom/Makefile
index a77071a..14ccba5 100644
--- a/app/ext_rom/Makefile
+++ b/app/ext_rom/Makefile
@@ -59,6 +59,7 @@
 	$(PREBOOT_ROOT_DIR)/lib/common/spi.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/spi_flash.o \
 	$(PREBOOT_ROOT_DIR)/lib/common/lib_common.o \
+	$(PREBOOT_ROOT_DIR)/lib/common/efuse.o \
 	$(HL_DEV_ARCH)/../common/security.o \
 	$(HL_DEV_ARCH)/$(HL_TARGET).o \
 	$(HL_DEV_ARCH)/../common/$(HL_TARGET).o \
diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 0ce45d1..b4ebc33 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -889,6 +889,13 @@
 {
 	u32 manf_mode, module_id_3, module_id_4;
 
+	/*
+	 * On HL225B cards, no debug features are allowed.
+	 * Enabling RMA on HL225B card, should not disable security
+	 */
+	if (preboot_check_bis_compliant_mode())
+		return false;
+
 	module_id_3 = gaudi2_gpio_read(GAUDI2_MOD_ID_3_GPIO);
 	module_id_4 = gaudi2_gpio_read(GAUDI2_MOD_ID_4_GPIO);
 	manf_mode = gaudi2_gpio_read(GAUDI2_MANF_MODE_GPIO);
